web_src/js/features/repo-diff.ts

```diff
@@ -219,30 +222,51 @@ function initRepoDiffShowMore() {
 async function onLocationHashChange() {
   // try to scroll to the target element by the current hash
   const currentHash = window.location.hash;
-  if (!currentHash.startsWith('#diff-') && !currentHash.startsWith('#issuecomment-')) return;
+  const issueCommentPrefix = '#issuecomment-';
+  const isDiffHash = currentHash.startsWith('#diff-');
+  const isIssueCommentHash = currentHash.startsWith(issueCommentPrefix);
+  if (!isDiffHash && !isIssueCommentHash) return;
 
   // avoid reentrance when we are changing the hash to scroll and trigger ":target" selection
-  const attrAutoScrollRunning = 'data-auto-scroll-running';
-  if (document.body.hasAttribute(attrAutoScrollRunning)) return;
+  if (document.body.hasAttribute(diffAutoScrollAttr)) return;
```

Comment:
Why it is called `diffAutoScrollAttr`? See the comment, the `attrAutoScrollRunning` was designed for "issue comment scrolling"